velox/type/TimestampUtil.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/base/Exceptions.h"
+
+namespace facebook::velox {
+/// A static class that holds helper functions for TIMESTAMP type.
+class TimestampUtil {
+ public:
+  static constexpr int kTmYearBase = 1900;
+  static constexpr int64_t kLeapYearOffset = 4000000000ll;
+
+  inline static bool isLeap(int64_t y) {
+    return y % 4 == 0 && (y % 100 != 0 || y % 400 == 0);
+  }
+
+  inline static int64_t leapThroughEndOf(int64_t y) {
+    // Add a large offset to make the calculation for negative years correct.
+    y += kLeapYearOffset;
+    VELOX_DCHECK_GE(y, 0);
+    return y / 4 - y / 100 + y / 400;
+  }
+
+  static constexpr int monthLengths[][12] = {
+      {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31},
+      {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31},
+  };
+
+  // clang-format off
+  static constexpr char intToStr[][3] = {
```

Comment:
private

Reply:
Fixed in https://github.com/facebookincubator/velox/pull/7246